src/NServiceBus.Metrics/ResetMetricTimer.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Features;
+
+class ResetMetricTimer
+{
+    public event EventHandler NoMessageSentForAWhile;
+
+    public ResetMetricTimer(FeatureConfigurationContext featureConfigurationContext)
+    {
+        timer = new System.Threading.Timer(ResetCounterValueIfNoMessageHasBeenProcessedRecently, null, 0, 2000);
+
+        featureConfigurationContext.Pipeline.OnReceivePipelineCompleted(e =>
+        {
+            DateTime timeSent;
+            if (e.TryGetTimeSent(out timeSent))
+            {
+                lastMessageProcessedTime = e.CompletedAt;
+            }
+
+            return Task.FromResult(0);
+        });
+
+    }
+
+    void ResetCounterValueIfNoMessageHasBeenProcessedRecently(object state)
+    {
+        // Concurreny : This runs every 2s and might just fire before an update
```

Comment:
I told @dvdstelt about the reading and writing on the field that you mentioned. We debated a little bit about it and came to the conclusion that probably it doesn't really matter here if we miss updates. We briefly talked about storing the timestamp as a long like you suggested but we did not think it was necessary. Performance counters had this in production for a while and so far it never raised any issues. we decided to not uses volatile or any other trickery. Happy to be convinced otherwise